forgerock-core/src/main/java/org/forgerock/android/auth/AppPinAuthenticator.kt

```diff
@@ -98,21 +100,30 @@ class AppPinAuthenticator(private val cryptoKey: CryptoKey,
         }
     }
 
+
     private fun generateCertificate(keyPair: KeyPair, subject: String): X509Certificate {
-        val validityBeginDate = Date()
-        val owner = X500Name("cn=$subject")
-        val sigAlgId: AlgorithmIdentifier =
-            DefaultSignatureAlgorithmIdentifierFinder().find("SHA256WithRSA")
-        val digAlgId: AlgorithmIdentifier = DefaultDigestAlgorithmIdentifierFinder().find(sigAlgId)
-        val privateKeyAsymKeyParam: AsymmetricKeyParameter =
-            PrivateKeyFactory.createKey(keyPair.private.encoded)
-        val sigGen = BcRSAContentSignerBuilder(sigAlgId, digAlgId).build(privateKeyAsymKeyParam)
-        val builder = X509v1CertificateBuilder(owner,
-            BigInteger.valueOf(System.currentTimeMillis()),
-            validityBeginDate,
-            validityBeginDate,
-            owner,
-            SubjectPublicKeyInfo.getInstance(keyPair.public.encoded))
-        return JcaX509CertificateConverter().getCertificate(builder.build(sigGen))
+        val initialLocale = Locale.getDefault()
```

Comment:
Instead of switching the global local, can we use this interface instead? org.spongycastle.cert.X509v1CertificateBuilder#X509v1CertificateBuilder(org.spongycastle.asn1.x500.X500Name, java.math.BigInteger, java.util.Date, java.util.Date, java.util.Locale, org.spongycastle.asn1.x500.X500Name, org.spongycastle.asn1.x509.SubjectPublicKeyInfo)